python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,1093 @@
+            point to the multiband objects.
+        """
+        singles = []
+        if image is None:
+            image = self.image
+        if mask is None:
+            mask = self.mask
+        if variance is None:
+            variance = self.variance
+
+        for n in range(len(image)):
+            dtype = self.singleType.__name__[-1]
+            imageType = getattr(afwMaskedImage, "MaskedImage"+dtype)
+            maskedImage = imageType(image=image[n], mask=mask[n], variance=variance[n])
+            single = self.singleType(maskedImage)
+            singles.append(single)
+        return tuple(singles)
+
+    @classmethod
+    def fromButler(cls, butler, filters, filterKwargs, *args, **kwargs):
+        """Load a multiband exposure from a butler
+
+        Because each band is stored in a separate exposure file,
+        this method can be used to load all of the exposures for
+        a given set of bands
+
+        Parameters
+        ----------
+        butler: `Butler`
+            Butler connection to use to load the single band
+            calibrated images
+        filters: list or str
+            List of filter names for each band
+        filterKwargs: dict
+            Keyword arguments to initialize a new instance of an
+            inherited class that are different for each filter.
+            The keys are the names of the arguments and the values
+            should also be dictionaries, with filter names as keys
+            and the value of the argument for the given filter as values.
+        args: list
+            Arguments to the Butler.
+        kwargs: dict
+            Keyword arguments to pass to initialize a new instance of an
+            inherited class that are the same in all bands.
+
+        Returns
+        -------
+        result: `MultibandExposure`
+            The new `MultibandExposure` created by combining all of the
+            single band exposures.
+        """
+        # Load the Exposure in each band
+        exposures = []
+        for f in filters:
+            if filterKwargs is not None:
+                for key, value in filterKwargs:
+                    kwargs[key] = value[f]
+            exposures.append(butler.get(*args, filter=f, **kwargs))
+        return cls(filters, exposures)
+
+    def setPsf(self, psf, filter):
+        """Set the PSF for a single Exposure
+
+        Parameters
+        ----------
+        psf: `meas.algorithms.coaddPsf`
+            The PSF to assign to the given exposure
+        filter: string or int
+            Either the index of the filter or name of the
+            filter for the `Exposure` in `self.singles`
+            to assign `psf`.
+        """
+        if isinstance(filter, str):
+            filter = self.filters.index(filter)
+        self.singles[filter].setPsf(psf)
+        # Clear the stored PSF image to be recalculated on demand later
+        self._psfImage = None
```

Comment:
Good catch. I removed the setPsf methods, no longer cache the result, and support passing either a `Point2D` or (x,y) position to use for PSF evaluation. I also changed the method to return a `MultibandImage`, since the user can easily convert this to an array.